### ISSUE TITLE
Windows with multiple GPUs keeps using DDP even with device = 'cpu' or device = '0'

### ISSUE THREAD

**Issue by SubhiH** [ultralytics/ultralytics#1947]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training, Detection, Multi-GPU

### Bug

<img width="1153" alt="Screenshot 2023-04-10 at 11 49 53 PM" src="https://user-images.githubusercontent.com/2580270/231051410-0252126b-6499-412d-b16e-83f6df157840.png">
<img width="1128" alt="Screenshot 2023-04-10 at 11 51 22 PM" src="https://user-images.githubusercontent.com/2580270/231051592-cdae4ed4-53a0-479f-9f69-fae5c3a2d1f7.png">

### Environment

- Ultralytics YOLOv8.0.73  Python-3.7.6 torch-1.7.0+cu110 CPU
- Windows 10 Pro
- Python 3.6.7
- Two GPUs RTX 3090

### Minimal Reproducible Example
Run the following on Windows 10 machine with two GPUs (RTX 3090 in my case):
`import ultralytics`
`from ultralytics import YOLO`
`ultralytics.checks()`
`model = YOLO('runs/detect/train2/weights/best.pt')`
`path = f'datasets/train2/data.yaml'`

`results = model.train(data=path, epochs=1000, batch = 8, imgsz = 960, device = 'cpu', dropout = 0.2, optimizer = 'Adam', fliplr=0, mixup=0, copy_paste = 0)`

### Additional

When I disconnect one of the gpus, everything works fine. I tried to enforce one GPU run in the code but it is still calling DDP which causing the issue.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@SubhiH 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by SubhiH** [comment]
@glenn-jocher 
Could you please help in with this issue?
Whenever I connect the two GPUs, I get DDP issue even though I set device=0 and latest version of the code. When I disconnect one of them, it works fine.
Please, any suggestions?
<img width="1470" alt="Screenshot 2023-05-16 at 8 16 50 PM" src="https://github.com/ultralytics/ultralytics/assets/2580270/71b733ff-d16c-4907-ba4b-80f319317d62">

**Comment by SubhiH** [comment]
Just an update, it seems **world_size** is being assigned as gpu count even if you send 0 or 'cpu'. And if world_size > 1, DDP will be enabled.
The temporary solution is to send the device = '' or None so it picks one of the gpus.
<img width="625" alt="Screenshot 2023-05-16 at 9 28 46 PM" src="https://github.com/ultralytics/ultralytics/assets/2580270/10eceefb-aa77-45c2-9c04-d4ae8aa88f2d">
